### PR TITLE
Grouping the mock transactions + folder structure fixes

### PR DESCRIPTION
**Pull request by mhlangagc**
- Grouped the Mock transactions by mock date values
- Grouped the TableView DataSource and Delegate functions separately. 
- Fixed the podfile to remove the installation errors.

### CODE REVIEW

File: MobileWallet/Screens/Home/TransactionHistory/TransactionsTableViewController.TableView.swift [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+import Foundation
+import UIKit
+
+extension TransactionsTableViewController {
+
+    // MARK: - Table view data source
+    override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let sectionHeaderView: UIView = UIView.init(frame: CGRect(x: 0, y: 0, width: tableView.frame.width, height: 56.0))
+        sectionHeaderView.backgroundColor = UIColor.white
```
[the remark this answers is not in the document]
Comment (Jasonvdb) [reply MobileWallet/Screens/Home/TransactionHistory/TransactionsTableViewController.TableView.swift]: Screenshot 2019-11-07 at 19 43 09

Would be better to make the whole view transparent with a background color on just the label. Similar to the value label in the transaction detail cell.

[the remark this answers is not in the document]
Comment (mhlangagc) [reply MobileWallet/Screens/Home/TransactionHistory/TransactionsTableViewController.TableView.swift]: It's probably better to make the whole view opaque with the white background (also consistent with other TableView structures e.g. your contacts). The label with a background colour seems out of place (If this is what you meant). 

IMG_DE8468C59B29-1

[the remark this answers is not in the document]
Comment (Jasonvdb) [reply MobileWallet/Screens/Home/TransactionHistory/TransactionsTableViewController.TableView.swift]: Ok maybe confirm with Karim. In the design videos the date separators are not sticky. But maybe that's a nice to have thing and Karim has some ideas on how to overlay it.

File: MobileWallet/Common/DateConfig.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+     */
+    class func getRelativeDayValue(fromDate date: Date) -> String? {
+        let dateFormatter = DateFormatter()
+
+        if Calendar.current.isDateInToday(date) {
+            return NSLocalizedString("Today", comment: "")
+        } else if Calendar.current.isDateInYesterday(date) {
+            return NSLocalizedString("Yesterday", comment: "")
+        } else {
+            dateFormatter.dateFormat = NSLocalizedString("MMM d, YYYY", comment: "")
```
Comment (Jasonvdb) [inline MobileWallet/Common/DateConfig.swift]: This date format string doesn't need to be translated, rather use a date format with the user's locale

File: MobileWallet/Common/DateConfig.swift [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+*/
+
+import Foundation
+
+final public class DateConfig {
+
+    /*
+     Creates a timestamp string value from a given date
+     */
+    class func getRelativeDayValue(fromDate date: Date) -> String? {
```
Comment (Jasonvdb) [inline MobileWallet/Common/DateConfig.swift]: Needs a unit test

File: MobileWallet/Common/DateConfig.swift [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        } else {
+            dateFormatter.dateFormat = NSLocalizedString("MMM d, YYYY", comment: "")
+            return dateFormatter.string(from: date)
+        }
+    }
+
+    /*
+     Checks if the given date falls in the current week
+     */
+    class func dateFallsInCurrentWeek(date: Date) -> Bool {
```
Comment (Jasonvdb) [inline MobileWallet/Common/DateConfig.swift]: Unit test

File: MobileWallet/Assets.xcassets/Colors/HomeBackground.colorset/Contents.json [hunk 5]
```diff
@@ -9,10 +9,10 @@
       "color" : {
         "color-space" : "srgb",
         "components" : {
-          "red" : "0.897",
+          "red" : "229",
```
Comment (Jasonvdb) [inline MobileWallet/Assets.xcassets/Colors/HomeBackground.colorset/Contents.json]: Is there an advantage to changing all the colours to 8-bit?

Comment (mhlangagc) [reply MobileWallet/Assets.xcassets/Colors/HomeBackground.colorset/Contents.json]: It's easier to read it from Zeplin mainly. No other advantage I know off technically.

File: MobileWallet/Screens/Home/TransactionHistory/TransactionsTableViewController.TableView.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (mhlangagc) [reply MobileWallet/Screens/Home/TransactionHistory/TransactionsTableViewController.TableView.swift]: 👍

File: MobileWallet/Assets.xcassets/Colors/HomeBackground.colorset/Contents.json (hunk 5 above)
Comment (Jasonvdb) [reply MobileWallet/Assets.xcassets/Colors/HomeBackground.colorset/Contents.json]: cool

File: MobileWallet/Screens/Home/TransactionHistory/TransactionsTableViewController.TableView.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (mhlangagc) [reply MobileWallet/Screens/Home/TransactionHistory/TransactionsTableViewController.TableView.swift]: I'll generate and send him a video of both designs.

File: MobileWalletTests/MobileWalletTests.swift [hunk 6]
```diff
@@ -91,7 +91,31 @@ class MobileWalletTests: XCTestCase {
...
+    func testDateFallsInWeek() {
+        let testDateString = "2019-10-01"
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-mm-dd"
+        guard let testDate = dateFormatter.date(from: testDateString) else {
+            XCTFail("Failed to get the date from the string value")
+            return
+        }
+        
+        if DateConfig.dateFallsInCurrentWeek(date: testDate) == true {
```
Comment (Jasonvdb) [inline MobileWalletTests/MobileWalletTests.swift]: I think this test will fail in a few days, `2019-10-01` will soon not be in the current week. Maybe adjust `dateFallsInCurrentWeek ` to accept 2 dates to compare if they're in the same week. Then in the implementation send through the current date as the second arg.

File: MobileWalletTests/MobileWalletTests.swift [hunk 7]
```diff
@@ -91,7 +91,31 @@ class MobileWalletTests: XCTestCase {
...
+            XCTFail("Date does not fall in current week. Test Date: \(testDate). Today's Date: \(Date())")
+        }
+    }
+    
+    
+    func testRelativeDayValue() {
+        let testDateString = Date()
+        let dateValue =  DateConfig.getRelativeDayValue(fromDate: testDateString)
+        if dateValue != NSLocalizedString("Today", comment: "") {
```
Comment (Jasonvdb) [inline MobileWalletTests/MobileWalletTests.swift]: Need to test all possible paths a function could execute. `Today`, `Tomorrow`, and full date

File: MobileWalletTests/MobileWalletTests.swift (hunk 6 above)
Comment (mhlangagc) [reply MobileWalletTests/MobileWalletTests.swift]: Ah! Good point.

File: MobileWalletTests/MobileWalletTests.swift [hunk 8]
```diff
@@ -92,79 +92,47 @@ class MobileWalletTests: XCTestCase {
...
-            return
-        }
-        let todaysPastTest = DateConfig.dateFallsInCurrentWeek(date: testDate)
-        XCTAssertEqual(todaysPastTest, false, "Test Failed. Date does not  fall in current week")
-        
+    func testDateFallsInCurrentWeek() {
+        /*
+         Test last week's date
+        */
+        let lastWeeksDate = Calendar.current.date(byAdding: .day, value: -7, to: Date())!
```
Comment (mhlangagc) [inline MobileWalletTests/MobileWalletTests.swift]: Should we keep these optionally unwrapped and not guarded?

**Comment by Jasonvdb** [comment]
Needed to sign commits to merge. https://github.com/tari-project/wallet-ios/pull/55

**Pull request closed without merging**